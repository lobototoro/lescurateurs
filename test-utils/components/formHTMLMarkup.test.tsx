import ArticleMarkupForm from "@/app/components/single-elements/articleHTMLForm";
import { fireEvent, render } from "@testing-library/react";
import { expect, it, vi } from "vitest";

const mockProps = {
    handleSubmit: vi.fn(),
    register: vi.fn(),
    errors: {},
    urlsToArray: [],
    updateUrls: vi.fn(),
    removeUrl: vi.fn(),
    addUrl: vi.fn(),
    handleChange: vi.fn(),
    addInputs: vi.fn(),
    removeInputs: vi.fn(),
    formSentModal: null,
    state: { message: false, text: '' },
    closeModal: vi.fn(),
    target: 'create',
    getMainContent: vi.fn(),
    setMainContent: vi.fn(),
  };

it('should render correct markup', () => {
  const { asFragment } = render(<ArticleMarkupForm {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});

it('should render correct markup for update', () => {
  const mockPropsUpdate = {
    ...mockProps,
    target: 'update',
  };

  const { asFragment, getByTestId } = render(
    <ArticleMarkupForm {...mockPropsUpdate} />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByTestId('title')).toHaveProperty('disabled', true);
});

it('Should display an additionnal link line with blinking validate button while modified', () => {
  const mockPropsAddUrl = {
    ...mockProps,
    urlsToArray: ['type: website, url: https://example.com, credits: test'],
    state: { message: true, text: 'TEST' },
    target: 'update',
  };
  const { getByTestId } = render(<ArticleMarkupForm {...mockPropsAddUrl} />);
  expect(getByTestId('select-type')).toBeDefined();
  fireEvent.change(getByTestId('select-type'), { target: { value: 'videos' } });
  expect(getByTestId('add-url-button')?.classList.toString()).toMatch(/blink/);
  fireEvent.click(getByTestId('add-url-button'));
  expect(getByTestId('add-url-button')?.classList.toString()).not.toMatch(/blink/);
});

