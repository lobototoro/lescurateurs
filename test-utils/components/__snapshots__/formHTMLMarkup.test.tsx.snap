// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correct markup 1`] = `
<DocumentFragment>
  <form
    data-testid="article-form"
  >
    <div
      class="mt-6"
    >
      <div
        class="field"
      >
        <label
          aria-labelledby="title"
          class="label"
          for="title"
        >
          Titre
        </label>
        <div
          class="field"
        >
          <input
            class="input"
            data-testid="title"
            id="title"
            type="text"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          aria-labelledby="introduction"
          class="label"
          for="introduction"
        >
          introduction
        </label>
        <textarea
          aria-label="introduction"
          class="textarea"
          data-testid="introduction"
          id="introduction"
          rows="5"
        />
      </div>
      <div
        class="field"
      >
        <label
          aria-label="main"
          class="label"
          for="main"
        >
          Texte
        </label>
      </div>
      <div
        class="field"
      >
        <label
          aria-label="urlToMainIllustration"
          class="label is-inline-flex"
          for="urlToMainIllustration"
        >
          lien vers l'illustration
        </label>
        <div
          class="control"
        >
          <input
            aria-label="urlToMainIllustration"
            class="input"
            data-testid="urlToMainIllustration"
            id="urlToMainIllustration"
            type="url"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          aria-label="mainAudioUrl"
          class="label"
          for="mainAudioUrl"
        >
          lien audio principal
        </label>
        <div
          class="control"
        >
          <input
            aria-label="mainAudioUrl"
            class="input"
            data-testid="mainAudioUrl"
            id="mainAudioUrl"
            type="url"
          />
        </div>
      </div>
      <div>
        <h4
          class="undefined article-title_medium article-title_white mt-5 mb-4"
        >
          Ajouter des liens multimédias
        </h4>
      </div>
      <div
        class="field"
        data-testid="url-inputs-container"
      >
        <div
          class="mt-5"
        >
          <div
            class="button is-primary is-size-6 has-text-white"
            data-testid="add-url"
          >
            +
          </div>
          <div
            class="button is-primary ml-2 is-size-6 has-text-white"
            data-testid="remove-url"
          >
            -
          </div>
        </div>
      </div>
    </div>
    <div
      class="field mt-5"
    >
      <input
        class="button is-primary is-size-6 has-text-white"
        data-testid="final-submit"
        type="submit"
        value="Valider"
      />
    </div>
  </form>
</DocumentFragment>
`;

exports[`should render correct markup for update 1`] = `
<DocumentFragment>
  <form
    data-testid="article-form"
  >
    <div
      class="mt-6"
    >
      <div
        class="field"
      >
        <label
          aria-labelledby="title"
          class="label"
          for="title"
        >
          Titre
        </label>
        <div
          class="field"
        >
          <input
            class="input"
            data-testid="title"
            disabled=""
            id="title"
            type="text"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          aria-labelledby="introduction"
          class="label"
          for="introduction"
        >
          introduction
        </label>
        <textarea
          aria-label="introduction"
          class="textarea"
          data-testid="introduction"
          id="introduction"
          rows="5"
        />
      </div>
      <div
        class="field"
      >
        <label
          aria-label="main"
          class="label"
          for="main"
        >
          Texte
        </label>
      </div>
      <div
        class="field"
      >
        <label
          aria-label="urlToMainIllustration"
          class="label is-inline-flex"
          for="urlToMainIllustration"
        >
          lien vers l'illustration
        </label>
        <div
          class="control"
        >
          <input
            aria-label="urlToMainIllustration"
            class="input"
            data-testid="urlToMainIllustration"
            id="urlToMainIllustration"
            type="url"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          aria-label="mainAudioUrl"
          class="label"
          for="mainAudioUrl"
        >
          lien audio principal
        </label>
        <div
          class="control"
        >
          <input
            aria-label="mainAudioUrl"
            class="input"
            data-testid="mainAudioUrl"
            id="mainAudioUrl"
            type="url"
          />
        </div>
      </div>
      <div>
        <h4
          class="undefined article-title_medium article-title_white mt-5 mb-4"
        >
          Ajouter des liens multimédias
        </h4>
      </div>
      <div
        class="field"
        data-testid="url-inputs-container"
      >
        <div
          class="mt-5"
        >
          <div
            class="button is-primary is-size-6 has-text-white"
            data-testid="add-url"
          >
            +
          </div>
          <div
            class="button is-primary ml-2 is-size-6 has-text-white"
            data-testid="remove-url"
          >
            -
          </div>
        </div>
      </div>
    </div>
    <div
      class="field mt-5"
    >
      <input
        class="button is-primary is-size-6 has-text-white"
        data-testid="final-submit"
        type="submit"
        value="Valider"
      />
    </div>
  </form>
</DocumentFragment>
`;
